cmd/kubeadm/app/util/config/common.go

```diff
@@ -243,6 +244,10 @@ func validateKnownGVKs(gvks []schema.GroupVersionKind) error {
 // Only kubeadm kinds are migrated.
 func MigrateOldConfig(oldConfig []byte, allowExperimental bool) ([]byte, error) {
 	newConfig := [][]byte{}
+	gv := kubeadmapiv1old.SchemeGroupVersion
+	if allowExperimental {
+		gv = kubeadmapiv1.SchemeGroupVersion
+	}
```

Comment:
can we move these lines near the call to MarshalKubeadmConfigObject()?

Reply:
Updated.